packages/playwright-core/src/utils/third_party/tar/patch.diff

```diff
@@ -0,0 +1,359 @@
+diff --git a/extract.js b/extract.js
+index 739ebc9..4375232 100644
+--- a/extract.js
++++ b/extract.js
+@@ -1,15 +1,13 @@
+-const { Writable, Readable, getStreamError } = require('streamx')
+-const FIFO = require('fast-fifo')
+-const b4a = require('b4a')
++const { Writable, Readable } = require('stream')
+ const headers = require('./headers')
+ 
+-const EMPTY = b4a.alloc(0)
++const EMPTY = Buffer.alloc(0)
+ 
+ class BufferList {
+   constructor () {
+     this.buffered = 0
+     this.shifted = 0
+-    this.queue = new FIFO()
++    this.queue = []
+ 
+     this._offset = 0
+   }
+@@ -38,11 +36,11 @@ class BufferList {
+       chunks.push(chunk)
+     }
+ 
+-    return b4a.concat(chunks)
++    return Buffer.concat(chunks)
+   }
+ 
+   _next (size) {
+-    const buf = this.queue.peek()
++    const buf = this.queue[0]
+     const rem = buf.byteLength - this._offset
+ 
+     if (size >= rem) {
+@@ -71,18 +69,17 @@ class Source extends Readable {
+     this._parent = self
+   }
+ 
+-  _read (cb) {
++  _read () {
+     if (this.header.size === 0) {
+       this.push(null)
+     }
+     if (this._parent._stream === this) {
+       this._parent._update()
+     }
+-    cb(null)
```

Comment:
Why did this go away?

Reply:
Before, this class extended `streamx.Readable`, but it now extends `node:stream.Readable`. The `_read` implementation is slightly different, see https://github.com/mafintosh/streamx/blob/be4bbc8ba0b4c862a3d996e484675d4b6297136d/README.md?plain=1#L136-L137. Luckily it was synchronous before, so we can remove this callback easily.